### PR TITLE
Correct svds dimension check.

### LINKED ISSUE #110: Starting Vector Initializing Fails For Right Singular Vectors
I try to initialize svds with the correct right singular vector,  but I get a DimensionMismatch() error.  The sizes should all be correct,  so there appears to be a problem with how this is verified.
```julia
using Arpack


# create a simple rank 1 matrix
Ml = 4
Nr = 8
u = randn(Ml) + 1im * randn(Ml)
v = randn(Nr) + 1im * randn(Nr)

Mat = u * v' 

# perform an svd to obtain the principal eigenvector
# But initialize the right singular value to the correct value
sv, stuff = svds(Mat, nsv=1, v0 = v)
```
I get the following errors:
```julia
julia> include("svdfail.jl")
ERROR: LoadError: DimensionMismatch("")
Stacktrace:
 [1] _eigs(::Arpack.AtA_or_AAt{Complex{Float64},Array{Complex{Float64},2}}, ::LinearAlgebra.UniformScaling{Bool}; nev::Int64, ncv::Int64, which::Symbol, tol::Float64, maxiter::Int64, sigma::Nothing, v0::Array{Complex{Float64},1}, ritzvec::Bool) at /home/matt/.julia/packages/Arpack/o35I5/src/Arpack.jl:123
 [2] #eigs#10 at /home/matt/.julia/packages/Arpack/o35I5/src/Arpack.jl:46 [inlined]
 [3] _svds(::Array{Complex{Float64},2}; nsv::Int64, ritzvec::Bool, tol::Float64, maxiter::Int64, ncv::Int64, v0::Array{Complex{Float64},1}) at /home/matt/.julia/packages/Arpack/o35I5/src/Arpack.jl:338
 [4] #svds#33 at /home/matt/.julia/packages/Arpack/o35I5/src/Arpack.jl:242 [inlined]
 [5] top-level scope at /mnt/WorkSpace/projects/Maestro/PulseWave/svdfail.jl:14
 [6] include(::String) at ./client.jl:457
 [7] top-level scope at REPL[4]:1
```

### PR DESCRIPTION
**Pull request by joshpoll**
Resolves #110.

### CODE REVIEW

**Comment by joshpoll** [comment]
I'm still pretty new to Julia, so I'm happy to add a regression test if I can be pointed in the right direction!

**Comment by ViralBShah** [comment]
Can you add a test?

**Comment by ViralBShah** [comment]
@andreasnoack Is this ok? 

@joshpoll It would be nice to add a test if possible.

File: src/Arpack.jl [hunk 1]
```diff
@@ -339,8 +339,9 @@ function _svds(X; nsv::Int = 6, ritzvec::Bool = true, tol::Float64 = 0.0, maxite
     end
     m, n = size(X)
     otype = eltype(X)
-    if length(v0) ∉ [0,n]
-        throw(DimensionMismatch("length of v0, the guess for the starting right Krylov vector, must be 0, or $n, got $(length(v0))"))
+    if length(v0) ∉ [0,min(m, n)]
```
Comment (andreasnoack) [inline src/Arpack.jl:342]:
```suggestion
    if length(v0) ∉ (0, min(m, n))
```
to avoid the vector allocation now that you are here anyway.

**Review by andreasnoack — COMMENTED** [review]
Yes. This should be good to go once a test is added.

**Comment by ViralBShah** [comment]
@joshpoll See this file for the existing tests and add something at the end of the file:

https://github.com/JuliaLinearAlgebra/Arpack.jl/blob/master/test/runtests.jl

**Comment by joshpoll** [comment]
Sorry I got sidetracked by the end of the semester! I'll add a test

File: test/runtests.jl [hunk 2]
```diff
@@ -413,3 +413,12 @@ end
     @test abs.(r.U[:, 1:n]'r0.U[:, 1:n]) ≈ I
     @test abs.(r.V[:, 1:n]'r0.V[:, 1:n]) ≈ I
 end
+
+# Regression test for #110.
```
Comment (andreasnoack) [inline test/runtests.jl:417]: It would be good with a versions that hits the error path as well

**Comment by joshpoll** [comment]
I'm not sure why this test is failing. It seems unrelated to the PR. A test that should be breaking is passing on nightly. I'm also having trouble reproducing these runs locally, since I'm not sure how to force Julia to use the local modified Arpack.

**Comment by andreasnoack** [comment]
Yes. I don't think it as anything to do with the changes here so let's consider that in a separate issue.

**Pull request merged**